src/parser.rs

```diff
@@ -758,6 +768,25 @@ fn parse_word_parts(
     )
   }
 
+  fn expand_tilde(result: &mut [WordPart]) {
+    if let Some(WordPart::Text(text)) = result.last_mut() {
+      if text.contains(" ~ ")
+        || text.contains("~/")
+        || (text.find('~').unwrap_or(0) == text.len() - 1)
+      {
+        let temp = text.clone().replace(
+          '~',
+          home_dir()
+            .unwrap_or(PathBuf::from("~"))
+            .to_str()
+            .unwrap_or("~"),
+        );
+        text.clear();
+        text.push_str(temp.as_str());
+      }
+    }
+  }
```

Comment:
I don't think we should do this while parsing, but instead during execution. The home directory in the shell might change while running. For example, the following test case should pass:

```sh
$ export HOME=test ; echo ~/test/~/test
test/test/~/test
```